vpr/src/place/timing_place.h

```diff
@@ -55,40 +97,148 @@ class PlacerCriticalities {
     float criticality(ClusterNetId net, int ipin) const { return timing_place_crit_[net][ipin]; }
 
-    //Returns the range of clustered netlist pins (i.e. ClusterPinIds) which were modified
-    //by the last call to update_criticalities()
+    /**
+     * @brief Returns the range of clustered netlist pins (i.e. ClusterPinIds) which
+     *        were modified by the last call to PlacerCriticalities::update_criticalities().
+     */
     pin_range pins_with_modified_criticality() const;
 
   public: //Modifiers
-    //Incrementally updates criticalities based on the atom netlist criticalitites provied by
-    //timing_info and the provided criticality_exponent.
+    /**
+     * @brief Updates criticalities based on the atom netlist criticalitites
+     *        provided by timing_info and the provided criticality_exponent.
+     */
     void update_criticalities(const SetupTimingInfo* timing_info, float criticality_exponent);
 
-    //Override the criticality of a particular connection
+    ///@brief Override the criticality of a particular connection.
     void set_criticality(ClusterNetId net, int ipin, float val);
 
+    ///@brief Set `update_enabled` to true.
+    void enable_update() { update_enabled = true; }
+
+    ///@brief Set `update_enabled` to true.
+    void disable_update() { update_enabled = false; }
+
   private: //Data
+    ///@brief The clb netlist in the placement context.
     const ClusteredNetlist& clb_nlist_;
-    const ClusteredPinAtomPinsLookup& pin_lookup_;
 
-    ClbNetPinsMatrix<float> timing_place_crit_; /* [0..cluster_ctx.clb_nlist.nets().size()-1][1..num_pins-1] */
+    ///@brief The lookup table that maps atom pins to clb pins.
+    const ClusteredPinAtomPinsLookup& pin_lookup_;
 
-    //The criticality exponent when update_criticalites() was last called (used to detect if incremental update can be used)
+    /**
+     * @brief The matrix that stores criticality value for each connection.
+     *
+     * Index range: [0..cluster_ctx.clb_nlist.nets().size()-1][1..num_pins-1]
+     */
+    ClbNetPinsMatrix<float> timing_place_crit_;
+
+    /**
+     * The criticality exponent when update_criticalites() was last called
+     * (used to detect if incremental update can be used).
+     */
     float last_crit_exponent_ = std::numeric_limits<float>::quiet_NaN();
 
-    //Set of pins with criticaltites modified by last call to update_criticalities()
+    ///@brief Set of pins with criticaltites modified by last call to update_criticalities().
     vtr::vec_id_set<ClusterPinId> cluster_pins_with_modified_criticality_;
+
+    ///@brief Updates criticalities: incremental V.S. from scratch
+    void incr_update_criticalities(const SetupTimingInfo* timing_info);
+    void recompute_criticalities();
+
+    ///@brief Flag that turns on/off the update_criticalities() routine.
+    bool update_enabled = true;
+
+    ///@brief Flag that checks if criticalities needs to be recomputed for all connections.
```

Comment:
Add "... i.e. they cannot be recomputed incrementally for only connections whose slacks have been updated in the last timing analysis"
(also, typo:  needs -> need)